Obsidian.API/Containers/AbstractContainer.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Obsidian.API
+{
+    public abstract class AbstractContainer : IEnumerable<ItemStack>, IEnumerable
```

Comment:
Not sure about having `Abstract` in the name. `Container`/`BaseContainer` perhaps?

`IEnumerable<T>` derives from `IEnumerable`